libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/NavEventsPopulateUtil.kt

```diff
@@ -0,0 +1,105 @@
+package com.mapbox.navigation.core.telemetry
+
+import com.mapbox.android.telemetry.TelemetryUtils
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.base.common.logger.model.Message
+import com.mapbox.base.common.logger.model.Tag
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.core.telemetry.events.AppMetadata
+import com.mapbox.navigation.core.telemetry.events.FeedbackEvent
+import com.mapbox.navigation.core.telemetry.events.NavigationEvent
+import com.mapbox.navigation.utils.internal.ifNonNull
+import com.mapbox.navigation.utils.internal.logD
+
+/**
+ * Populate base Navigation Event
+ *
+ * @param sdkIdentifier sdk identifier
+ * @param originalRoute original route (optional)
+ * @param routeProgress route progress (optional)
+ * @param lastLocation last location
+ * @param locationEngineNameExternal location engine name
+ * @param percentTimeInPortrait [ApplicationLifecycleMonitor.obtainPortraitPercentage]
+ * @param percentTimeInForeground [ApplicationLifecycleMonitor.obtainForegroundPercentage]
+ * @param navigatorSessionIdentifier nav session id (identifier of instance of MapboxNavigation)
+ * @param driverModeId id of driver mode (FreeDrive or ActiveGuidance)
+ * @param driverMode one of [FeedbackEvent.DriverMode]
+ * @param driverModeStartTime driver mode start time.
+ * Use [TelemetryUtils.generateCreateDateFormatted]
+ * @param rerouteCount reroute count
+ * @param eventVersion events version [MapboxNavigationTelemetry.EVENT_VERSION]
+ * @param appMetadata use [MapboxNavigationTelemetry.createAppMetadata]
+ */
+internal fun NavigationEvent.populate(
+    sdkIdentifier: String,
+    originalRoute: DirectionsRoute?,
+    routeProgress: RouteProgress?,
+    lastLocation: Point?,
+    locationEngineNameExternal: String?,
+    percentTimeInPortrait: Int?,
+    percentTimeInForeground: Int?,
+    navigatorSessionIdentifier: String?,
+    driverModeId: String?,
+    @FeedbackEvent.DriverMode driverMode: String?,
+    driverModeStartTime: String?,
+    rerouteCount: Int?,
+    eventVersion: Int,
+    appMetadata: AppMetadata?,
+) {
+    logD(Tag("MapboxNavigationTelemetry"), Message("populateNavigationEvent"))
+
+    this.sdkIdentifier = sdkIdentifier
+
+    ifNonNull(routeProgress) { routeProgressNonNull ->
+        stepIndex = routeProgressNonNull.currentLegProgress?.currentStepProgress?.stepIndex ?: 0
```

Comment:
Is `stepIndex` a required value in the schema? If not, let's not populate it when it's not available. We should steer away from default values in telemetry, only report what we actually have and know.

This also applies to all default values below.

Reply:
I'd not change it there, this's copy-paste from another place. we must review all schemas/events with https://github.com/mapbox/mapbox-navigation-android/issues/4744